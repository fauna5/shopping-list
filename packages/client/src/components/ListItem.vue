<template>
  <div class="list-item" >
    <input type="checkbox" :checked="complete" v-on:click="updateStatus"/>
    <h2 :class="{strikethrough: this.complete}">{{this.title}}</h2>
  </div>
</template>

<script>
import { UPDATE } from '../store/actions';
export default {
  name: 'list-item',
  props: {
    title: {
      type: String,
      required: true,
    },
    complete: {
      type: Boolean,
      required: true,
    },
    id: {
      type: Number,
      required: true,
    },
  },
  methods: {
    updateStatus() {
      this.$store.dispatch(UPDATE, { id: this.id, complete: !this.complete });
    },
  },
};
</script>

<style scoped>
h2 {
  margin: 0px;
}
.list-item {
  display: flex;
  align-items: center;
  padding: 10px;
  border: 1px lightgray solid;
  margin: 4px 0px;
}
.list-item:hover {
  background: #eeeeee;
}
.list-item > input {
  transform: scale(1.4);
  cursor: pointer;
}
.list-item > h2 {
  font-size: 20px;
  font-weight: normal;
  padding-left: 10px;
}
.strikethrough {
  text-decoration: line-through;
}
</style>