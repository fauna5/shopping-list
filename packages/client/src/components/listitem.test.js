import Vuex from 'vuex';
import { shallowMount, createLocalVue } from '@vue/test-utils';
import createStore from '../store/store';

import ListItem from './ListItem';

const localVue = createLocalVue();
localVue.use(Vuex);

describe('ListItem', () => {
  it('renders', () => {
    const wrapper = shallowMount(ListItem, {
      store: createStore(),
      propsData: {
        id: 2,
        title: 'bread',
        complete: false,
      },
      localVue,
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('renders with strikethrough when complete', () => {
    const wrapper = shallowMount(ListItem, {
      store: createStore(),
      propsData: {
        id: 2,
        title: 'bread',
        complete: true,
      },
      localVue,
    });
    expect(wrapper.find('h2').classes()).toContain('strikethrough');
  });

  it('should dispatch item update on click', () => {
    const updateMutation = jest.fn();

    const wrapper = shallowMount(ListItem, {
      store: new Vuex.Store({
        actions: { UPDATE: updateMutation },
      }),
      propsData: {
        id: 2,
        title: 'bread',
        complete: false,
      },
      localVue,
    });
    const checkbox = wrapper.find('input');
    checkbox.trigger('click');
    expect(updateMutation).toHaveBeenCalledWith(expect.anything(), { complete: true, id: 2 }, undefined);
  });
});
