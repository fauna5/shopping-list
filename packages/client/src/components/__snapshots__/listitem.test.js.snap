// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem renders 1`] = `
<div
  class="list-item"
>
  <input
    type="checkbox"
  />
   
  <h2
    class=""
  >
    bread
  </h2>
</div>
`;
